activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -1595,6 +1602,12 @@ def arel(aliases = nil) # :nodoc:
       @arel ||= with_connection { |c| build_arel(c, aliases) }
     end
 
+    # Returns a Common Table Expression (CTE) from the relation to be
+    # used directly as input to +.with+.
+    def as_cte(subquery_alias, materialized: nil)
+      Arel::Nodes::Cte.new(subquery_alias, arel, materialized:)
+    end
```

Comment:
Two thoughts:

1. Given that the `#arel` method (which returns an `Arel::SelectManager`) is marked `:nodoc:`, I'd guess that this should be too.
2. I wonder if the word "arel" should be in the method name, as in `#as_arel_cte` or maybe just `#arel_cte` for consistency with `#arel`.

Reply:
@rafaelfranca are we okay with moving Arel to the public API in a case like this?